driver/src/models.rs

```diff
@@ -21,9 +21,30 @@ pub trait Serializable {
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
 #[serde(rename_all = "camelCase")]
 pub struct State {
-  pub state_hash: String,
-  pub state_index: i32,
-  pub balances: Vec<u128>,
+    pub state_hash: String,
+    pub state_index: i32,
+    balances: Vec<u128>,
+}
+
+impl State {
+    pub fn new(state_hash: String, state_index: i32, balances: Vec<u128>) -> Self {
+        State { state_hash, state_index, balances }
+    }
+    pub fn balance_index(token_id: u8, account_id: u16) -> usize {
+        TOKENS as usize * account_id as usize  + token_id as usize
+    }
+    pub fn read_balance(&self, token_id: u8, account_id: u16) -> u128 {
+        self.balances[State::balance_index(token_id, account_id)]
+    }
+    pub fn increment_balance(&mut self, token_id: u8, account_id: u16, amount: u128) {
+        self.balances[State::balance_index(token_id, account_id)] += amount;
+    }
+    pub fn decrement_balance(&mut self, token_id: u8, account_id: u16, amount: u128) {
+        self.balances[State::balance_index(token_id, account_id)] -= amount;
+    }
+    pub fn read_balances(&self) -> Vec<u128> {
```

Comment:
I am not sure I understand...

Reply:
If I read correctly, the only reason this is needed, is because we want to serialize the balances. We could also expose a `serialize_balances` method directly as part of this trait.

Otherwise, I'd suggest to return a &Vec<u128> here to avoid the clone. 

Or am I missing a use case?

Reply:
Ok I see what you mean now. This is because `read_balances` is only used once (in the same place as `serialize_balances` could be used